assets/src/js/Components/ArchivePicker/ArchivePicker.js

```diff
@@ -356,6 +364,12 @@ export default function ArchivePicker({view = ADMIN_VIEW}) {
     });
   };
 
+  const setCustomFeaturedImage = async imgId => {
+    await wp.data.dispatch('core/editor').editPost({featured_media: imgId});
+    dispatch({type: ACTIONS.CLOSE_SIDEBAR});
+    document.querySelector('.media-modal-close').click();
```

Comment:
This behaviour should work for all instances, but I have witnessed situations too where it does not work so I will try something different and see

Reply:
This still doesn't work as expected for the Media & Text block 😕

Reply:
Funny thing is, I haven't changed the implementation yet 😅. [This](https://github.com/greenpeace/planet4-master-theme/blob/main/assets/src/js/Components/ArchivePicker/ArchivePicker.js#L316) is the implantation for the Media and Text block, same as Image Block and Featured Image.

Reply:
What I see now for the Media & Text block is that the Greenpeace Media sidebar closes, but not the whole modal. Same for the Image block.

Also, when clicking on `Import to Library & Post` for a brief moment I see an `Add image to post` button, which is confusing because after a few seconds it disappears and an `Added!` message shows. So there's actually no need for that button.

<img width="362" alt="Screenshot 2024-10-03 at 10 22 30" src="https://github.com/user-attachments/assets/7fdfcf7b-8543-4816-adab-2f99a38a0e55">

Reply:
@Osong-Michael I still see this